src/languages/gml.js

```diff
@@ -2994,135 +937,572 @@ export default function(hljs) {
+        endScope: "string",
+        contains: [
+          STRING_ESCAPE,
+          STRING_SUBSTITUTION,
+          {
+            match: /[^\n"{]/,
+            scope: "string"
+          }
+        ]
+      },
+      {
+        scope: "string",
+        begin: "@'",
+        end: "'",
+        contains: [STRING_NUMERICAL_SUBSTITUTION]
+      },
+      {
+        scope: "string",
+        begin: "@\"",
+        end: "\"",
+        contains: [STRING_NUMERICAL_SUBSTITUTION]
+      },
+      {
+        scope: "string",
+        begin: /"/,
+        end: /"/,
+        illegal: "\\n",
+        contains: [
+          STRING_ESCAPE, 
+          STRING_NUMERICAL_SUBSTITUTION
+        ]
+      }
+    ]
+  };
+
+  /**
+   * Various representations of numbers
+   */
+  const NUMBER = {
+    className: "literal",
+    variants: [
+      { match: /(\B|^)\$[0-9a-fA-F]+/ },
+      { match: /(\B|^)#[0-9a-fA-F]+/ },
+      { match: /\b0x[0-9a-fA-F][0-9a-fA-F_]*/ },
+      { match: /\b0b[01][01_]*/ },
+      { match: /\b[0-9][0-9_.]*/ }
+    ]
+  };
+
+  const COMMENT_LINE_INNER = {
+    match: /[^\n]+/,
+    className: "comment"
+  };
+
+  /**
+   * Pre-processor modes for macro definitions and regions.
+   */
+  const PREPROCESSOR = {
+    end: /$/,
+    variants: [
+      {
+        begin: [
+          /#macro\s+/,
+          VALID_IDENTIFIER_REG
+        ],
+        className: {
+          1: "keyword",
+          2: "literal"
+        },
+        contains: lateConcat(
+          EXPRESSION,
+          [
+            COMMENT,
+            {
+              match: /\\\n/
+            }
+          ]
+        ),
+      },
+      {
+        begin: "#define"
+      },
+      {
+        begin: /#(end)?region\b/,
+        beginScope: "keyword",
+        contains: [COMMENT_LINE_INNER]
+      },
+    ]
+  };
+
+  /**
+   * Dot accessor usage with a special highlighting case for `global`.
+   */
+  const PROP_ACCESS = [
+    {
+      match: [
+        "global",
+        DOT_ACCESSOR_REG,
+        VALID_IDENTIFIER_REG
+      ],
+      className: {
+        1: "literal",
+        3: "property"
+      }
+    },
+    {
+      match: [
+        DOT_ACCESSOR_REG,
+        VALID_IDENTIFIER_REG,
+        /\s*\(/
+      ],
+      className: {
+        2: "function"
+      }
+    },
+    {
+      match: [
+        DOT_ACCESSOR_REG,
+        VALID_IDENTIFIER_REG
+      ],
+      className: {
+        2: "property"
+      }
+    },
+  ];
+
+  /**
+   * Function call sites, just looking for `<ident>(`. This creates false positives
+   * for keywords such as `if (<condition>)`, so has lower priority in the mode `contains` list.
+   */
+  const FUNCTION_CALL = {
+    begin: [
+      VALID_IDENTIFIER_REG,
+      /\s*?/,
+      /\(/
+    ],
+    className: {
+      1: "function"
+    }
+  };
+
+  /**
+   * The manual likes using `obj_` and such to define assets. Sneaky trick to make it look nicer :P
+   */
+  const USER_ASSET_CONSTANT = {
+    className: "literal",
```

Comment:
Do these refer to some sort of variable/object?  If so I think `variable` would be correct.  If they can't be redefined then perhaps `variable.constant`.

We typically use `literal` only to refer to literal values that "are themselves", such as:

- null
- true
- false

Ie, most grammars have very few things scoped as `literal` - typically a very short list of keywords.  Most places this grammar uses literal another scope should likely be preferred.  Did you see our official scope list?

Reply:
I've swapped that out for `variable.constant`. I was not aware of the proper full scopes list earlier as I was just working off of what was present in `gml.css` in the GM Manual initially, I've gone ahead and had a proper look over the scopes list and made various fixes here and there.